Panic on complex embedded structures

The following program `sample.go` triggers a panic:

```go
package main

import "github.com/traefik/yaegi/interp"

const src = `package foo

type TypeA struct {
	B TypeB
}

type TypeB struct {
	C1 *TypeC
	C2 *TypeC
}

type TypeC struct {
	Val string
	D *TypeD
	D2 *TypeD
}

type TypeD struct {
	Name        string
}

func build() *TypeA {
	return &TypeA{
		B: TypeB{
			C2: &TypeC{Val: "22"},
		},
	}
}
func Bar(s string) string { 
	a := build()
	return s + "-" + a.B.C2.Val
}
`

func main() {
	i := interp.New(interp.Options{})

	_, err := i.Eval(src)
	if err != nil {
		panic(err)
	}

	v, err := i.Eval("foo.Bar")
	if err != nil {
		panic(err)
	}

	bar := v.Interface().(func(string) string)

	r := bar("Kung")
	println(r)
}
```

Expected result:
```console
$ go run ./sample.go
// output
Kung-22
```

Got:
```console
$ go run ./sample.go
// output
25:9: panic
30:7: panic
panic: reflect.Set: value of type struct { C1 *struct { Val string; D *struct { Name string }; D2 *struct { Name string } }; C2 *struct { Val string; D *struct { Name string }; D2 *struct { Name string } } } is not assignable to type struct { C1 *interface {}; C2 *interface {} } [recovered]
        panic: reflect.Set: value of type struct { C1 *struct { Val string; D *struct { Name string }; D2 *struct { Name string } }; C2 *struct { Val string; D *struct { Name string }; D2 *struct { Name string } } } is not assignable to type struct { C1 *interface {}; C2 *interface {} } [recovered]
        panic: reflect.Set: value of type struct { C1 *struct { Val string; D *struct { Name string }; D2 *struct { Name string } }; C2 *struct { Val string; D *struct { Name string }; D2 *struct { Name string } } } is not assignable to type struct { C1 *interface {}; C2 *interface {} }
```

--- a/_test/issue-1007.go
+++ b/_test/issue-1007.go
@@ -1,0 +1,40 @@
+package main
+
+type TypeA struct {
+	B TypeB
+}
+
+type TypeB struct {
+	C1 *TypeC
+	C2 *TypeC
+}
+
+type TypeC struct {
+	Val string
+	D   *TypeD
+	D2  *TypeD
+}
+
+type TypeD struct {
+	Name string
+}
+
+func build() *TypeA {
+	return &TypeA{
+		B: TypeB{
+			C2: &TypeC{Val: "22"},
+		},
+	}
+}
+
+func Bar(s string) string {
+	a := build()
+	return s + "-" + a.B.C2.Val
+}
+
+func main() {
+	println(Bar("test"))
+}
+
+// Output:
+// test-22

--- a/interp/type.go
+++ b/interp/type.go
@@ -1551,7 +1551,7 @@ func hasRecursiveStruct(t *itype, defined map[string]*itype) bool {
 		defined[typ.path+"/"+typ.name] = typ
 
 		for _, f := range typ.field {
-			if hasRecursiveStruct(f.typ, defined) {
+			if hasRecursiveStruct(f.typ, copyDefined(defined)) {
 				return true
 			}
 		}

